src/main/java/org/springframework/integration/dsl/context/IntegrationFlowContext.java

```diff
@@ -149,8 +149,11 @@ public void register(String flowId, IntegrationFlow integrationFlow, boolean aut
 	 */
 	public synchronized void remove(String flowId) {
 		if (this.registry.containsKey(flowId)) {
+			IntegrationFlow theFlow = this.registry.remove(flowId);
+			if (theFlow instanceof Lifecycle) {
+				((Lifecycle) theFlow).stop();
+			}
```

Comment:
Agreed, I just re-checked the `start()/stop()` contracts.

Will merge after my call ends.